politeiawww/prices.go

```diff
@@ -12,33 +13,69 @@ import (
 	database "github.com/decred/politeia/politeiawww/cmsdatabase"
 )
 
+const binanceURL = "https://api.binance.com"
 const poloURL = "https://poloniex.com/public"
 const httpTimeout = time.Second * 3
 const pricePeriod = 900
 
+const dcrSymbolBinance = "DCRBTC"
+const usdtSymbolBinance = "BTCUSDT"
+
+const dcrSymbolPolo = "BTC_DCR"
+const usdtSymbolPolo = "USDT_BTC"
+
 type poloChartData struct {
 	Date            uint64  `json:"date"`
 	WeightedAverage float64 `json:"weightedAverage"`
 }
 
+// Set the last date to use polo as 4/1/2019.  If any start date requested is
+// after that date then use Binance instead.
+var endPoloDate = time.Date(2019, 4, 1, 0, 0, 0, 0, time.UTC)
+
 // GetMonthAverage returns the average USD/DCR price for a given month
 func (p *politeiawww) GetMonthAverage(month time.Month, year int) (uint, error) {
 	startTime := time.Date(year, month, 1, 0, 0, 0, 0, time.UTC)
 	endTime := startTime.AddDate(0, 1, 0)
 
+	if time.Now().Before(endTime) {
+		return 0, fmt.Errorf("Requested rate end time (%v) is past current time (%v)", endTime, time.Now())
+	}
+
 	unixStart := startTime.Unix()
 	unixEnd := endTime.Unix()
 
-	// Download BTC/DCR and USDT/BTC prices from Poloniex
-	dcrPrices, err := getPrices("BTC_DCR", unixStart, unixEnd)
-	if err != nil {
-		return 0, err
-	}
-	btcPrices, err := getPrices("USDT_BTC", unixStart, unixEnd)
-	if err != nil {
-		return 0, err
-	}
+	var dcrPrices map[uint64]float64
+	var btcPrices map[uint64]float64
+	var err error
+
+	// Use Binance if start date is AFTER 3/31/19
+	if startTime.Before(endPoloDate) {
+		// Download BTC/DCR and USDT/BTC prices from Polo
+		dcrPrices, err = getPricesPolo(dcrSymbolPolo, unixStart, unixEnd)
+		if err != nil {
+			log.Error(err)
```

Comment:
Don't log an error message without giving it context.  You are also logging this error message twice.  It gets logged here and then then again in the calling function `processInvoiceExchangeRate`.

The way that I would suggest doing this is returning errors here (and below), then logging the error in the calling function.  Returning a more descriptive error would also be helpful so that you can know exactly what the error is referring to.

```
return 0, fmt.Errorf("getPricesPolo %v: %v", dcrSymbolPolo, err)
```